src/pages/index.js

```diff
@@ -20,88 +87,176 @@ const cardSection = new Section(
   ".gallery__cards"
 );
 
-// Render initial cards
-cardSection.renderItems();
+function fillCardLike(){
+  const likeBtn = document.querySelector('.card__like-button');
+  likeBtn.classList.toggle('card__like-button--active');
+}
 
-//---------------------FORMS AND SUBMITS--------------------------
+function emptyCardLike(){
+  const likeBtn = document.querySelector('.card__like-button');
+  likeBtn.classList.remove('card__like-button--active');
+}
+
+// Fetch API cards and render them inside the same section
+api.getInitialCards()
+.then((cards) => {
+  cards.forEach((cardData) => {
+    const cardElement = createCard(cardData); // Generate card using createCard
+    cardSection.addItem(cardElement); // Properly add to `.gallery__cards`
+    
+    console.log(cardData.isLiked)
+    if(cardData.isLiked){
+      console.log(cardData._id);  
+      fillCardLike();
+    }else{
+      emptyCardLike();
+    }
+  });
+});
+
+//---------------------FORMS AND VALIDATION AND SUBMIT BUTTONS RESPONSES--------------------------
 
 // Initialize form validation
 const addCardValidator = new FormValidator(config, document.querySelector("#add__modal_form"));
 const profileValidator = new FormValidator(config, document.querySelector("#edit__modal_form"));
+const avatarValidator = new FormValidator(config, document.querySelector("#avatar__modal_form"));
 addCardValidator.enableValidation();
 profileValidator.enableValidation();
+avatarValidator.enableValidation();
 
 //-----------------------POPUPS--------------------------------
-
 // Create Popups
 const imagePopup = new PopupWithImage("#modal__picture");
 imagePopup.setEventListeners();
 
 //  Initialize `UserInfo`
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
-  jobSelector: ".profile__description"
+  jobSelector: ".profile__description",
+  avatarSelector: ".profile__avatar"
 });
 
 const profilePopup = new PopupWithForm("#profileModal", (formData) => {
-  userInfo.setUserInfo(formData);
+  userInfo.setUserInfo(formData); // Make changes on the page right away
+  
+  api.updateUserInfo(formData.title, formData.about) // sends to Api.js to save to database
+    .then((updatedData) => {
+      profilePopup.close(); 
+    })
+    .catch((err) => {
+      console.error("Error updating profile:", err);
+    });
```

Comment:
Every server request in `index.js` should look something like this:

```js
api.someMethod()
.then(res => {
// here is the response that should be used to update the DOM.
// also in `then` you should close the popup
})
.catch(err => // here you catch possible errors)
.finally(() => // here you return the default button text back )
``` 

Please, correct it in the whole code